### PR TITLE
Add PlayerArmorStandCreateEvent

### PR DESCRIPTION
**Pull request by MisterVector**
This PR adds an event when a player creates an armor stand.

### CODE REVIEW

**Comment by kashike** [comment]
...and you can't use an interact event... why?

**Comment by MisterVector** [comment]
Aren't there events already present that could be replaced by an interact event? I could use an interact event, but isn't specificity good? Your comment is making me think though.

File: Spigot-API-Patches/0051-Add-PlayerArmorStandCreateEvent.patch [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
++import org.bukkit.event.HandlerList;
++import org.bukkit.event.player.PlayerEvent;
++
++/**
++ * Called when a player is about to create an armor stand
++ */
++public class PlayerArmorStandCreateEvent extends PlayerEvent implements Cancellable {
++    private static HandlerList handlers = new HandlerList();
++    private ArmorStand stand;
++    private boolean canceled;
```
Comment (kashike) [inline Spigot-API-Patches/0051-Add-PlayerArmorStandCreateEvent.patch]: Should be `cancelled`.

**Review by kashike — CHANGES_REQUESTED** [review]

**Comment by kashike** [comment]
Yes and no. What's the use case for this specific event?

**Comment by AlfieC** [comment]
Should this not be abstracted and a destroy event added too?

**Comment by MisterVector** [comment]
@kashike you can retrieve the location the armor stand will be placed without doing some calculations ahead of time. On top of that, it brings this in line with an event like PlayerArmorStandManipulateEvent, keeping consistency.

@AlfieC one thing at a time :) but I can totally see that.

File: Spigot-API-Patches/0051-Add-PlayerArmorStandCreateEvent.patch (hunk 1 above)
Comment (MisterVector) [reply Spigot-API-Patches/0051-Add-PlayerArmorStandCreateEvent.patch]: Fixed.

**Comment by zachbr** [comment]
The gist from a quick discussion in IRC is that this event is unnecessary given that it's covered by other events and, while it does expose a few extra items, this is not the only way to get to them.

Therefore we will decline, thanks anyway.

**Pull request closed without merging**